wp_api/src/request/endpoint/posts_endpoint.rs

```diff
@@ -66,11 +74,289 @@ mod tests {
         validate_wp_v2_endpoint(endpoint.delete(&PostId(54)), "/posts/54?force=true");
     }
 
+    #[rstest]
+    #[case(PostListParams::default(), "")]
+    #[case(generate!(PostListParams, (page, Some(2))), "page=2")]
+    #[case(generate!(PostListParams, (per_page, Some(2))), "per_page=2")]
+    #[case(generate!(PostListParams, (search, Some("foo".to_string()))), "search=foo")]
+    #[case(generate!(PostListParams, (after, Some("2023-08-14 17:00:00.000".to_string()))), "after=2023-08-14+17%3A00%3A00.000")]
+    #[case(generate!(PostListParams, (modified_after, Some("2023-08-14 17:00:00.000".to_string()))), "modified_after=2023-08-14+17%3A00%3A00.000")]
```

Comment:
Once we have the `Date` stuff working, this is another opportunity to use that class – it'd be super nice to be able to provide an object to this instead and be guaranteed that the formatting is correct.

Reply:
I mentioned this on Slack, but this is already handled in the PR I am working on that adds the `Date` type. 👍